### PR TITLE
libevhtp: Update to 1.2.0

### PR DESCRIPTION
**Pull request by neheb**
In the tree, libevhtp is used only for Seafile. This is the most recent
version that Haiwen lists as working.

Updated threads patch for OpenSSL 1.1.

Removed Maintainer as he is inactive.

Simplified Makefile a little bit and reorganized for consistency between
packages.

Fixed compilation with uClibc-ng.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

Maintainer: none
Compile tested: arc700

### CODE REVIEW

File: libs/libevhtp/Makefile [hunk 1]
```diff
@@ -11,25 +11,28 @@
...
 PKG_SOURCE_URL:=https://codeload.github.com/criticalstack/libevhtp/tar.gz/$(PKG_VERSION)?
-PKG_HASH:=725dd0a32237b2a097cdc2003d09278a082ccce310aaba7441a5fd8fa8f26635
+PKG_HASH:=d8d98072693f5f68ccd74d327dedfa2f6add4446ac2799689c2f58dd480aa301
+
+PKG_MAINTAINER:=
+PKG_LICENSE:=BSD-3-Clause
+PKG_LICENSE_FILES:=LICENSE
 
-PKG_INSTALL:=1
+CMAKE_INSTALL:=1
```
Comment (diizzyy) [inline libs/libevhtp/Makefile]: Why is this added?

Comment (neheb) [reply libs/libevhtp/Makefile]: Allows removal of the InstallDev section.

Comment (diizzyy) [reply libs/libevhtp/Makefile]: I personally think that should be kept

Comment (neheb) [reply libs/libevhtp/Makefile]: git grep CMAKE_INSTALL\: | wc -l
33
git grep cmake.mk | wc -l
68

No clear winner. I prefer CMAKE_INSTALL as it simplifies the Makefile.

Comment (cotequeiroz) [reply libs/libevhtp/Makefile]: I agree with @neheb here.

Comment (diizzyy) [reply libs/libevhtp/Makefile]: Unless I'm reading the code wrong it'll pull in everything (running a regular install) which will lead to possible build bloat(?) and imho obfuscation.

Comment (neheb) [reply libs/libevhtp/Makefile]: It installs everything from ipkg-install to the staging_dir, yes. But disk space is not really a concern for the host. Only for the device running the library. But Build/Install is not touched.

**Comment by cotequeiroz** [comment]
I'm not sure what's best to do with the seafile packages.  We aren't using the latest version of libevhtp because of them.
Apparently the oniguruma version bundled with libevhtp had many unfixed CVEs, so they decided to unbundle it:
https://github.com/criticalstack/libevhtp/commit/a31601cfd1b6198c10c577dc40b6fd5df492eafe
Then, there's the python 2.7 removal looming, and python3 compatibility being listed under "long term" in the seafile roadmap--as I understood, it is post 8.0, which is expected in 2020/04:
https://www.seafile.com/en/roadmap/
I've looked at https://github.com/haiwen/seafile-server/issues/67#issuecomment-337904800, which has a recipe for making seafile work with more recent versions of libevhtp.  I've checked the sources, and they still need the patches.  Note that I did not even try to apply them.  I expect it to need more rigorous testing, but the seafile packages are currently unmaintained here.
If we are to touch any of this, I think we should at least build a recent oniguruma library and use it instead of the one bundled with libevhtp.

File: libs/libevhtp/Makefile (hunk 1 above)
Comment (cotequeiroz) [reply libs/libevhtp/Makefile]: They install exactly the same files:
```
$ ls -R ipkg-install/
ipkg-install/:
usr

ipkg-install/usr:
include  lib

ipkg-install/usr/include:
evhtp.h  evthr.h  htparse.h

ipkg-install/usr/lib:
libevhtp.a
```

Comment (diizzyy) [reply libs/libevhtp/Makefile]: @cotequeiroz 
I was referring in general terms, I doubt it'll be true for all.

**Comment by cotequeiroz** [comment]
I just stumbled upon this:
https://github.com/criticalstack/libevhtp/issues/70

It seems to fix the problem that keeps us from updating this.  I just need to find the time to test it.

**Comment by neheb** [comment]
I believe I tried the seafile pull request. I gave up after I couldn't get it to compile properly. You might have better luck.

Anyway, I think this can be merged for now. It additionally fixes compilation without deprecated APIs (moot since python-cryptography requires them).
